src/View.elm

```diff
@@ -0,0 +1,178 @@
+                , button [ onClick (Edit NotEditing) ] [ text "discard" ]
+                ]
+
+        ( Nothing, [], Id id user ) ->
+            footer [ css [ backgroundColor (hex "eea") ] ]
+                [ button [ onClick (PutUser id user) ] [ text "save" ]
+                , button [ onClick (Edit NotEditing) ] [ text "discard" ]
+                , button [ onClick (IntendDeleteUser (Just ( id, user ))) ] [ text "delete" ]
+                ]
+
+        _ ->
+            footer [] []
+
+
+viewCreateUserMedallion : UnderEdit -> Html Msg
+viewCreateUserMedallion underEdit =
+    case underEdit of
+        NewUser user ->
+            medallion [ css [ backgroundColor (hex "aea") ] ] (inputs user (Edit << NewUser))
+
+        _ ->
+            medallion [ css [ cursor pointer ], onClick (Edit (NewUser initUser)) ] [ span [ css [ fontSize (rem 2) ] ] [ text "➕" ] ]
+
+
+viewUser : (List (Attribute Msg) -> List (Html Msg) -> Html Msg) -> UnderEdit -> ( Int, User ) -> Html Msg
+viewUser wrapper underEdit ( id, { firstName, lastName, elmSkill } as user ) =
+    let
+        default =
+            wrapper [ css [ cursor pointer ], onClick (Edit (Id id user)) ] (texts user)
+    in
+    case underEdit of
+        Id id_ user_ ->
+            if id == id_ then
+                wrapper [ css [ backgroundColor (hex "eea") ] ] (inputs user_ (Edit << Id id))
+
+            else
+                default
+
+        _ ->
+            default
+
+
+medallion : List (Attribute msg) -> List (Html msg) -> Html msg
+medallion extraAttrs =
+    div
+        (css
+            [ borderRadius (pct 50)
+            , border2 (px 1) solid
+            , displayFlex
+            , flexDirection column
+            , alignItems center
+            , justifyContent center
+            , property "align-content" "center"
+            , property "aspect-ratio" "1"
+            ]
+            :: extraAttrs
+        )
+
+
+row : List (Attribute msg) -> List (Html msg) -> Html msg
+row extraAttrs =
+    div (css [ borderBottom2 (px 1) solid, displayFlex, flexDirection column, padding (px 2) ] :: extraAttrs)
+
+
+footer : List (Attribute msg) -> List (Html msg) -> Html msg
+footer extraAttrs =
+    div (css [ displayFlex, alignItems center, padding (px 8), position fixed, left (px 0), right (px 0), bottom (px 0), height (px 28) ] :: extraAttrs)
+
+
+modalFooter : List (Attribute msg) -> List (Html msg) -> Html msg
+modalFooter extraAttrs elems =
+    div []
+        [ div [ css [ position fixed, opacity (num 0.5), backgroundColor (hex "000"), left (px 0), right (px 0), bottom (px 0), top (px 0) ] ] []
+        , footer extraAttrs elems
+        ]
+
+
+inputs : User -> (User -> Msg) -> List (Html Msg)
+inputs ({ firstName, lastName, elmSkill } as user) toMsg =
+    [ div []
+        [ input [ css [ width (px 72) ], value firstName, placeholder "first name", onInput (\new -> { user | firstName = new } |> toMsg) ] []
```

Comment:
OK with non-CSS lists (value, placeholder, onInput) this gets really painful for me 😅

Reply:
I hear you. Hurt no more!